File: src/test/java/br/com/controllers/ProdutoFornecedorControllerTest.java
package br.com.controllers;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import br.com.newObjects.model.entities.classes.NovoProdutoFornecedor;

@DataJpaTest
@AutoConfigureTestDatabase(replace = Replace.NONE)
public class ProdutoFornecedorControllerTest {

    @Autowired
    private ProdutoFornecedorController produtoFornecedorController;
    
    @Test
    void testFind() {

    }

    @Test
    void testFindAll() throws Exception {
        for(int i = 0;i<4;i++){
            produtoFornecedorController.save(NovoProdutoFornecedor.novoProdutoFornecedor());
        }

        var a = produtoFornecedorController.findAll();
        a.forEach(produtoFornecedor -> System.out.println(produtoFornecedor.toString()));
        a.forEach(produtoFornecedor -> System.out.println(produtoFornecedor.getProduto()));
        a.forEach(produtoFornecedor -> System.out.println(produtoFornecedor.getFornecedor()));
    }

    @Test
    void testFindById() {

    }

    @Test
    void testRemove() {

    }

    @Test
    void testSave() {

    }

    @Test
    void testUpdate() {

    }
}

File: src/test/java/br/com/controllers/ProdutoFornecedorControllerTest.kt
package br.com.controllers

import br.com.newObjects.model.entities.classes.NovoProdutoFornecedor.novoProdutoFornecedor
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest

@DataJpaTest
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
class ProdutoFornecedorControllerTest(@Autowired private val produtoFornecedorController: ProdutoFornecedorController) {
    @Test
    fun testFind() {
    }

    @Test
    @Throws(Exception::class)
    fun testFindAll() {
        for (i in 0..3) {
            produtoFornecedorController.save(novoProdutoFornecedor())
        }
        val a = produtoFornecedorController.findAll()
        a.forEach { produtoFornecedor -> println(produtoFornecedor.toString()) }
        a.forEach { produtoFornecedor -> println(produtoFornecedor.produto) }
        a.forEach { produtoFornecedor -> println(produtoFornecedor.fornecedor) }
    }

    @Test
    fun testFindById() {
    }

    @Test
    fun testRemove() {
    }

    @Test
    fun testSave() {
    }

    @Test
    fun testUpdate() {
    }
}
